reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md

```diff
@@ -0,0 +1,682 @@
+to the **ICommandPredictor** interface. For more information about creating a command line
+predictor, see [How to create a command line predictor][04].
+
+The following code implements the methods necessary to add predictor behavior to your feedback
+provider.
+
+- `CanAcceptFeedback()` - Overloads the method of the **ICommandPredictor** interface. This method
+  returns a boolean value that indicates whether the predictor accepts a specific type of feedback.
+- `GetSuggestion()` - Overloads the method of the **ICommandPredictor** interface. This method
+  returns a `SuggestionPackage` object that contains the suggestions to be displayed by the
+  predictor.
+- `OnCommandLineAccepted()` - Overloads the method of the **ICommandPredictor** interface. This
+  method is called when a command line is accepted to execute.
+
+```csharp
+#region ICommandPredictor
+
+// Gets a value indicating whether the predictor accepts a specific kind of feedback.
+// client - Represents the client that initiates the call.
+// feedback - the specific kind of feedback
+public bool CanAcceptFeedback(PredictionClient client, PredictorFeedbackKind feedback)
+{
+    return feedback switch
+    {
+        PredictorFeedbackKind.CommandLineAccepted => true,
+        _ => false,
+    };
+}
+
+// Get the predictive suggestions. It indicates the start of a suggestion rendering session.
+// client - Represents the client that initiates the call.
+// context - The PredictionContext object to be used for prediction.
+// cancellationToken - The cancellation token to cancel the prediction.
+public SuggestionPackage GetSuggestion(PredictionClient client, PredictionContext context,
+    CancellationToken cancellationToken)
+{
+    if (_candidates is not null)
+    {
+        string input = context.InputAst.Extent.Text;
+        List<PredictiveSuggestion>? result = null;
+
+        foreach (string c in _candidates)
+        {
+            if (c.StartsWith(input, StringComparison.OrdinalIgnoreCase))
+            {
+                result ??= new List<PredictiveSuggestion>(_candidates.Count);
+                result.Add(new PredictiveSuggestion(c));
+            }
+        }
+
+        if (result is not null)
+        {
+            return new SuggestionPackage(result);
+        }
+    }
+
+    return default;
+}
+
+// A command line was accepted to execute. The predictor can start processing early as needed
+// with the latest history.
+public void OnCommandLineAccepted(PredictionClient client, IReadOnlyList<string> history)
+{
+    // Reset the candidate state once the command is accepted.
+    _candidates = null;
+}
+
+public void OnSuggestionDisplayed(PredictionClient client, uint session, int countOrIndex) { }
+
+public void OnSuggestionAccepted(PredictionClient client, uint session, string acceptedSuggestion) { }
+
+public void OnCommandLineExecuted(PredictionClient client, string commandLine, bool success) { }
```

Comment:
I believe you don't need to implement them now. The interface has default implementation for them.